pkgs/tools/typesetting/asciidoc/default.nix

```diff
@@ -218,7 +218,7 @@ in python3.pkgs.buildPythonApplication rec {
     # cannot find their neighbours (e.g. pdflatex doesn't find mktextfm).
     # We can remove PATH= when those impurities are fixed.
     # TODO: Is this still necessary when using texlive?
-    sed -e "s|^ENV =.*|ENV = dict(XML_CATALOG_FILES='${docbook_xml_dtd_45}/xml/dtd/docbook/catalog.xml ${docbook_xsl_ns}/xml/xsl/docbook/catalog.xml ${docbook_xsl}/xml/xsl/docbook/catalog.xml', PATH='${lib.makeBinPath [ texlive coreutils gnused ]}')|" \
+    sed \#-e "s|^ENV =.*|ENV = dict(XML_CATALOG_FILES='${docbook_xml_dtd_45}/xml/dtd/docbook/catalog.xml ${docbook_xsl_ns}/xml/xsl/docbook/catalog.xml ${docbook_xsl}/xml/xsl/docbook/catalog.xml', PATH='${lib.makeBinPath [ texlive coreutils gnused ]}', **(dict(filter(lambda v: v[0] == 'SOURCE_DATE_EPOCH', os.environ.items()))))|" \
```

Comment:
What's this "\#"? (I've never seen it before.) And sed supports inline Python?

Reply:
> What's this "#"? (I've never seen it before.)

Oops, that was not supposed to be there :)

> And sed supports inline Python?

No, but sed is used there to tweak a Python file, so this is evaluated at run time - which is what we want, since that way a2x will apply SOURCE_DATE_EPOCH when it was set by whoever invoked it

Reply:
Ah, thanks for explaining :-)